packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -56,7 +56,7 @@ function NavigationMenuSelector( {
 		return (
 			navigationMenus?.map( ( { id, title }, index ) => {
 				const label =
-					decodeEntities( title.rendered ) ||
+					decodeEntities( title?.rendered ) ||
```

Comment:
Bug fix: if there is no `rendered` property _at all_ we need to handle that and ensure we get a fallback title.